src/com/hms/flexyosisoftconnector/OSIsoftServer.java

```diff
@@ -392,8 +392,12 @@ public int initTags() throws JSONException {
         PayloadBuilder.initWebIdList();
 
         for (int i = 0; i < TagInfoManager.getTagInfoList().size(); i++) {
-          res = setTagWebId((TagInfo) TagInfoManager.getTagInfoList().get(i));
-          if (res != NO_ERROR) return retval = res;
+          TagInfo tag = (TagInfo) TagInfoManager.getTagInfoList().get(i);
+
+          if (tag != null) {
+            res = setTagWebId(tag);
+            if (res != NO_ERROR) return retval = res;

```

Comment:
why are you returning here instead of breaking?

Reply:
I moved that part of code that since it's the same as what it used to be without thinking about it. I guess if the tag doesn't have a web ID set everything else because pretty pointless, but a break would make it so only those particular tags will not be able to get into OSIsoft.

The better thing to do is break and log a warning for that tag instead returning and having the following tags also not work, right?

Reply:
I just missed this the first time this code came through review. The break would be breaking out of the for loop not the case. Single returns per method are easier to understand and follow. Also im not a fan of inline conditionals.

Reply:
For the other thing, it'll just keep going instead of ruining the tags after now. I think the logic makes more sense like this.